### PR TITLE
#1112 - web optimizations

### PR DESCRIPTION
**Pull request by proddy**
Make the Web pack smaller by
- replace notistack with react-toast
- removing the bold font (was only used in teble headers)
- change minify to terser
- add HTML minifying as a backup
- other minor fixes

in total saves 25.5KB in flash

### CODE REVIEW

File: interface/src/project/SettingsCustomization.tsx [hunk 1]
```diff
@@ -507,61 +503,59 @@ const SettingsCustomization: FC = () => {
...
+                            type="deleted"
+                            isSet={(de.m & DeviceEntityMask.DV_DELETED) === DeviceEntityMask.DV_DELETED}
+                          />
+                        </ToggleButton>
+                      </ToggleButtonGroup>
                     </Cell>
-                    <Cell>{!deviceEntity && formatName(de)}</Cell>
-                    <Cell>{!deviceEntity && !(de.m & DeviceEntityMask.DV_READONLY) && formatValue(de.mi)}</Cell>
-                    <Cell>{!deviceEntity && !(de.m & DeviceEntityMask.DV_READONLY) && formatValue(de.ma)}</Cell>
-                    <Cell>{!deviceEntity && formatValue(de.v)}</Cell>
```
Comment (MichaelDvP) [inline interface/src/project/SettingsCustomization.tsx:564]: On my old laptop it takes ~5 seconds for each character input in dialog if the list is rendered in background. With empty list i can just type a new name without waiting.

**Review by MichaelDvP — COMMENTED** [review]
Tried to compile and upload, but can not access the webpage, body of all pages seems empty.
Will test with a new branch...

File: interface/src/project/SettingsCustomization.tsx (hunk 1 above)
Comment (proddy) [reply interface/src/project/SettingsCustomization.tsx:564]: ok, I wondered why that clause was in there. I'll see if I can optimize it

Comment (MichaelDvP) [reply interface/src/project/SettingsCustomization.tsx:564]: Not rendering the table jumps to top of table after editing, the trick was to render the table, but not the content of the cells. Then scolling stays at the same place.

Comment (proddy) [reply interface/src/project/SettingsCustomization.tsx:564]: i know, but I'd like to code it properly. the biggest problem with React is that the DOM and all its children are re-rendered when a state changes (in this case the deviceEntity). The solution is to use a pattern like https://www.developerway.com/posts/react-re-renders-guide for the Modals where the props and states are separated. Some coding to do....

**Comment by MichaelDvP** [comment]
Now it works, After reboot/upload the snackbar is red but says "Upload successfull".

**Comment by proddy** [comment]
> Now it works, After reboot/upload the snackbar is red but says "Upload successfull". Te difference is ~17kB, mainly the size of the font, zipped 11kB. But it gives a little room for v3.5.1.

fixed that. The last dev with notistack v3 used 2026865 bytes, so its around 25K if you count that version upgrade. Still not bad for 60 mins of work

**Pull request merged**